weave

```diff
@@ -96,15 +96,21 @@ usage() {
 }
 
 docker_sock_volume_mount() {
-  if [ -z "$DOCKER_HOST" ]; then
-      DOCKER_SOCK="/var/run/docker.sock"
-  elif echo "$DOCKER_HOST" | grep -q "^unix://" >/dev/null; then
+  if echo "$DOCKER_HOST" | grep -q "^unix://" >/dev/null; then
       DOCKER_SOCK="${DOCKER_HOST#unix://}"
+  else
+      DOCKER_SOCK="/var/run/docker.sock"
   fi
-  [ -z "$DOCKER_SOCK" ] || echo "-v $DOCKER_SOCK:$DOCKER_SOCK"
+  echo "-v $DOCKER_SOCK:$DOCKER_SOCK"
 }
 
 exec_remote() {
+    # Pass through DOCKER_HOST if it is a Unix socket;
+    # a TCP socket may be secured by TLS, in which case we can't use it
+    if echo "$DOCKER_HOST" | grep -q "^unix://" ; then
```

Comment:
I hesitated to make the change so broad; `docker_sock_volume_mount` is called in a number of places.

However, if we don't want to pass in `$DOCKER_HOST` in some places, maybe we don't want to pass it in in all places.